#DPATCHLEVEL=1
--- arj-3.10.18.orig/gnu/makefile.in	2003-05-14 02:28:28.000000000 -0700
+++ arj-3.10.18/gnu/makefile.in	2003-09-23 13:04:43.000000000 -0700
@@ -110,7 +110,7 @@
 	$(CC) $(COPT) -o$@ $<
 
 $(ARJCRYPT_DIR)/%.o: $(BASEDIR)/%.c
-	$(CC) $(COPT) -o$@ $<
+	$(CC) $(COPT) -fPIC -o$@ $<
 
 $(SFXSTUB_DIR)/%.o: $(BASEDIR)/%.c
 	$(CC) $(COPT) -o$@ $<
@@ -140,7 +140,7 @@
 	$(CC) $(ARJSFXJR_COPT) -o$@ $<
 
 $(ARJCRYPT_DIR)/%.o: $(SRC_DIR)/%.c
-	$(CC) $(ARJCRYPT_COPT) -o$@ $<
+	$(CC) $(ARJCRYPT_COPT) -fPIC -o$@ $<
 
 $(REARJ_DIR)/%.o: $(SRC_DIR)/%.c
 	$(CC) $(REARJ_COPT) -o$@ $<


