#DPATCHLEVEL=1
--- arj-3.10.18.orig/gnu/configure.in
+++ arj-3.10.18/gnu/configure.in
@@ -80,6 +80,15 @@
 sparc*)
         AC_DEFINE(ALIGN_POINTERS)
         ;;
+ia64*)
+       AC_DEFINE(ALIGN_POINTERS)
+       ;;
+arm*)
+       AC_DEFINE(ALIGN_POINTERS)
+       ;;
+hppa*)
+       AC_DEFINE(ALIGN_POINTERS)
+       ;;
 esac

 dnl Registration wizard is named "arj-register" under UNIX-like platforms, to
