#DPATCHLEVEL=1
diff -Naur arj-3.10.19/gnu/configure.in arj-3.10.19.patched/gnu/configure.in
--- arj-3.10.19/gnu/configure.in	2004-01-25 01:39:36.000000000 +0100
+++ arj-3.10.19.patched/gnu/configure.in	2004-04-02 05:57:23.000000000 +0200
@@ -43,7 +43,7 @@
 DLL_FLAGS="-shared"
 REQUIRES_DEF=
 OS_ID="UNIX"
-CFLAGS="$CFLAGS -O2 -D_UNIX"
+OS_DEF="-D_UNIX"
 
 case $host_os in
 linux*)
@@ -65,11 +65,11 @@
 os2*)
 	PROG_EXT=".exe"
 	DLL_EXT=".dll"
-	CFLAGS="$CFLAGS -O2 -D_OS2"
 	LDFLAGS="-s"
 	DLL_FLAGS="-Zdll -s"
 	REQUIRES_DEF=1
 	OS_ID="OS2"
+	OS_DEF="-D_OS2"
 	;;
 esac
 
@@ -159,6 +159,7 @@
 AC_SUBST(DLL_CFLAGS)
 AC_SUBST(REQUIRES_DEF)
 AC_SUBST(OS_ID)
+AC_SUBST(OS_DEF)
 AC_SUBST(LIBS)
 AC_SUBST(DYN_LIBS)
 AC_SUBST(LD_STRIP)
diff -Naur arj-3.10.19/gnu/makefile.in arj-3.10.19.patched/gnu/makefile.in
--- arj-3.10.19/gnu/makefile.in	2004-01-25 01:39:36.000000000 +0100
+++ arj-3.10.19.patched/gnu/makefile.in	2004-04-02 05:57:01.000000000 +0200
@@ -38,25 +38,25 @@
 LOCALE = en
 endif
 
-CFLAGS_DEF = -DLOCALE=LANG_$(LOCALE) -DLOCALE_DESC="\"$(LOCALE)\"" \
-	     -DPKGLIBDIR="\"$(pkglibdir)\""
+ALL_CFLAGS = -DLOCALE=LANG_$(LOCALE) -DLOCALE_DESC="\"$(LOCALE)\"" \
+	     -DPKGLIBDIR="\"$(pkglibdir)\"" @OS_DEF@
 
 ifndef COMMERCIAL
 PACKAGE = s
 else
 PACKAGE = c
-CFLAGS_SW = -DCOMMERCIAL
+ALL_CFLAGS += -DCOMMERCIAL
 endif
 
 ifdef DEBUG
 DEBUG_SM = d
-CFLAGS_DBG = -g -DDEBUG
+ALL_CFLAGS += -g -DDEBUG
 else
 DEBUG_SM = r
 ADD_LDFLAGS = -s @LD_STRIP@
 endif
 
-CFLAGS = $(CFLAGS_SW) $(CFLAGS_DBG) $(CFLAGS_DEF)
+ALL_CFLAGS += $(CFLAGS)
 
 # Build Installation paths
 
@@ -84,7 +84,7 @@
 DLL_CFLAGS = @DLL_CFLAGS@
 REQUIRES_DEF = @REQUIRES_DEF@
 
-COPT = -c -I@OUT_DIR@ -I$(BASEDIR) -I$(SRC_DIR) @CFLAGS@ $(CFLAGS)
+COPT = -c -I@OUT_DIR@ -I$(BASEDIR) -I$(SRC_DIR) @CFLAGS@ $(ALL_CFLAGS)
 STD_COPT = -DARJUTIL $(COPT)
 ARJ_COPT = -DSFL=4 $(COPT)
 ARJSFXV_COPT = -DSFL=3 $(COPT)
@@ -214,22 +214,22 @@
 		filemode.o packager.o arjdata.o)
 
 $(TOOLS_DIR)/msgbind$x: $(MSGBIND_OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(MSGBIND_OBJS) $(LIBS)
+	$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o $@ $(MSGBIND_OBJS) $(LIBS)
 
 $(TOOLS_DIR)/today$x: $(TODAY_OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(TODAY_OBJS) $(LIBS)
+	$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o $@ $(TODAY_OBJS) $(LIBS)
 
 $(TOOLS_DIR)/make_key$x: $(MAKE_KEY_OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(MAKE_KEY_OBJS) $(LIBS)
+	$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o $@ $(MAKE_KEY_OBJS) $(LIBS)
 
 $(TOOLS_DIR)/postproc$x: $(POSTPROC_OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(POSTPROC_OBJS) $(LIBS)
+	$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o $@ $(POSTPROC_OBJS) $(LIBS)
 
 $(TOOLS_DIR)/join$x: $(JOIN_OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(JOIN_OBJS) $(LIBS)
+	$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o $@ $(JOIN_OBJS) $(LIBS)
 
 $(TOOLS_DIR)/packager$x: $(PACKAGER_OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(PACKAGER_OBJS) $(LIBS)
+	$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o $@ $(PACKAGER_OBJS) $(LIBS)
 
 # This rule tells that timestamping badly needs date_sig.c (it may bail out
 # otherwise when the timestamps are being created for the first time)
@@ -252,7 +252,7 @@
 		integr.o gost.o gost_t.o nmsg_crp.o arjcrypt.o)
 
 $(ARJCRYPT_DIR)/arjcrypt$d: $(ARJCRYPT_OBJS) $(TOOLS_DIR)/postproc$x
-	$(CC) $(CFLAGS) $(DLL_FLAGS) -o $@ $(ARJCRYPT_OBJS) $(ARJCRYPT_DEF) $(LIBS)
+	$(CC) $(ALL_CFLAGS) $(DLL_FLAGS) -o $@ $(ARJCRYPT_OBJS) $(ARJCRYPT_DEF) $(LIBS)
 	$(TOOLS_DIR)/postproc $@
 
 $(BASEDIR)/nmsg_crp.c: $(TOOLS_DIR)/msgbind$x $(RESFILE)
@@ -266,7 +266,7 @@
 	       nmsg_stb.o sfxstub.o)
 
 $(SFXSTUB_DIR)/sfxstub$x: $(SFXSTUB_OBJS) $(TOOLS_DIR)/postproc$x
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(SFXSTUB_OBJS) $(LIBS)
+	$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o $@ $(SFXSTUB_OBJS) $(LIBS)
 	$(TOOLS_DIR)/postproc$x $@ -sfx
 
 $(BASEDIR)/nmsg_stb.c: $(TOOLS_DIR)/msgbind$x $(RESFILE)
@@ -284,7 +284,7 @@
 	       arjsec_h.o arjsec_l.o garble.o scrnio.o ansi.o externs.o)
 
 $(ARJSFXV_DIR)/arjsfxv$x: $(ARJSFXV_OBJS) $(TOOLS_DIR)/postproc$x
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ARJSFXV_OBJS) $(LIBS) $(DYN_LIBS)
+	$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o $@ $(ARJSFXV_OBJS) $(LIBS) $(DYN_LIBS)
 	$(TOOLS_DIR)/postproc$x $@ -sfx
 
 $(BASEDIR)/fmsg_sfv.c $(BASEDIR)/imsg_sfv.c $(BASEDIR)/nmsg_sfv.c: $(TOOLS_DIR)/msgbind$x $(RESFILE)
@@ -301,7 +301,7 @@
 	      chk_fmsg.o arjsec_h.o arjsec_l.o garble.o externs.o)
 
 $(ARJSFX_DIR)/arjsfx$x: $(ARJSFX_OBJS) $(TOOLS_DIR)/postproc$x
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ARJSFX_OBJS) $(LIBS)
+	$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o $@ $(ARJSFX_OBJS) $(LIBS)
 	$(TOOLS_DIR)/postproc$x $@ -sfx
 
 $(BASEDIR)/fmsg_sfx.c $(BASEDIR)/imsg_sfx.c $(BASEDIR)/nmsg_sfx.c: $(TOOLS_DIR)/msgbind$x $(RESFILE)
@@ -316,7 +316,7 @@
 		crc32.o environ.o)
 
 $(ARJSFXJR_DIR)/arjsfxjr$x: $(ARJSFXJR_OBJS) $(TOOLS_DIR)/postproc$x
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ARJSFXJR_OBJS) $(LIBS)
+	$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o $@ $(ARJSFXJR_OBJS) $(LIBS)
 	$(TOOLS_DIR)/postproc$x $@ -sfx
 
 $(BASEDIR)/fmsg_sfj.c $(BASEDIR)/imsg_sfj.c $(BASEDIR)/nmsg_sfj.c: $(TOOLS_DIR)/msgbind$x $(RESFILE)
@@ -344,7 +344,7 @@
 		  $(ARJSFXV_DIR)/arjsfxv$x \
 		  $(ARJSFX_DIR)/arjsfx$x \
 		  $(ARJSFXJR_DIR)/arjsfxjr$x
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ARJ_OBJS) $(LIBS) $(DYN_LIBS)
+	$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o $@ $(ARJ_OBJS) $(LIBS) $(DYN_LIBS)
 	$(TOOLS_DIR)/join $(ARJ_DIR)/arj$x $(ARJSFXJR_DIR)/arjsfxjr$x
 	$(TOOLS_DIR)/join $(ARJ_DIR)/arj$x $(ARJSFX_DIR)/arjsfx$x
 	$(TOOLS_DIR)/join $(ARJ_DIR)/arj$x $(ARJSFXV_DIR)/arjsfxv$x
@@ -369,7 +369,7 @@
 
 $(REARJ_DIR)/rearj$x: $(REARJ_OBJS) \
 		      $(TOOLS_DIR)/postproc$x
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(REARJ_OBJS) $(LIBS)
+	$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o $@ $(REARJ_OBJS) $(LIBS)
 	$(TOOLS_DIR)/postproc $@
 
 $(BASEDIR)/fmsg_rej.c $(BASEDIR)/imsg_rej.c $(BASEDIR)/nmsg_rej.c: $(TOOLS_DIR)/msgbind$x $(RESFILE)
@@ -385,7 +385,7 @@
 
 $(REGISTER_DIR)/$(REGISTER)$x: $(REGISTER_OBJS) \
 		            $(TOOLS_DIR)/postproc$x
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(REGISTER_OBJS) $(LIBS)
+	$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o $@ $(REGISTER_OBJS) $(LIBS)
 	$(TOOLS_DIR)/postproc $@ -sfx
 
 $(BASEDIR)/fmsg_reg.c $(BASEDIR)/imsg_reg.c $(BASEDIR)/nmsg_reg.c: $(TOOLS_DIR)/msgbind$x $(RESFILE)
@@ -400,7 +400,7 @@
 	       scrnio.o)
 
 $(ARJDISP_DIR)/arjdisp$x: $(ARJDISP_OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ARJDISP_OBJS) $(LIBS)
+	$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o $@ $(ARJDISP_OBJS) $(LIBS)
 
 $(BASEDIR)/fmsg_adi.c $(BASEDIR)/imsg_adi.c $(BASEDIR)/nmsg_adi.c: $(TOOLS_DIR)/msgbind$x $(RESFILE)
 	$(TOOLS_DIR)/msgbind $(RESFILE) msg_adi $(OS_ID) $(PACKAGE) $(LOCALE) $(BASEDIR)
